data/list/perm.lean

```diff
@@ -429,6 +429,28 @@ theorem subperm_cons (a : α) {l₁ l₂ : list α} : a::l₁ <+~ a::l₂ ↔ l
   { exact ⟨u, perm_cons_inv p, s'⟩ }
 end, λ ⟨l, p, s⟩, ⟨a::l, skip a p, s.cons2 _ _ _⟩⟩
 
+theorem subperm_cons_left {a : α} {l₁ l₂ : list α} (d₁ : nodup l₁) (h₁ : a ∉ l₁) (h₂ : a ∈ l₂)
+ (s : l₁ <+~ l₂) : a :: l₁ <+~ l₂ :=
+begin
+  rcases s with ⟨l, p, s⟩,
+  induction s generalizing l₁,
+  case list.sublist.slnil { cases h₂ },
+  case list.sublist.cons : r₁ r₂ b s' ih {
+    simp at h₂,
+    cases h₂ with e m,
+    { subst b, exact ⟨a::r₁, skip a p, s'.cons2 _ _ _⟩ },
+    { rcases ih m d₁ h₁ p with ⟨t, p', s'⟩, exact ⟨t, p', s'.cons _ _ _⟩ } },
+  case list.sublist.cons2 : r₁ r₂ b s' ih {
+    have bm : b ∈ l₁ := (perm_subset p $ mem_cons_self _ _),
+    have am : a ∈ r₂ := h₂.resolve_left (λ e, h₁ $ e.symm ▸ bm),
+    rcases mem_split bm with ⟨t₁, t₂, rfl⟩,
+    have st : t₁ ++ t₂ <+ t₁ ++ b :: t₂ := by simp,
+    rcases ih am (nodup_of_sublist st d₁)
+                 (mt (λ x, subset_of_sublist st x) h₁)
+                 (perm_cons_inv $ p.trans perm_middle) with ⟨t, p', s'⟩,
```

Comment:
Why the extra indentation? It's not a meaningful alignment point - it's not even aligned to the first argument. The usual wrapping is an extra two space indentation, and that's what was in the original proof.